packages/terra-form-select/src/SelectField.jsx

```diff
@@ -112,60 +119,111 @@ const defaultProps = {
   variant: 'default',
 };
 
-const SelectField = ({
-  children,
-  defaultValue,
-  error,
-  help,
-  hideRequired,
-  isInline,
-  isInvalid,
-  isLabelHidden,
-  label,
-  labelAttrs,
-  maxWidth,
-  onChange,
-  placeholder,
-  required,
-  selectAttrs,
-  selectId,
-  showOptional,
-  value,
-  variant,
-  ...customProps
-}) => (
-  <Field
-    {...customProps}
-    label={label}
-    labelAttrs={labelAttrs}
-    error={error}
-    help={help}
-    hideRequired={hideRequired}
-    required={required}
-    showOptional={showOptional}
-    isInvalid={isInvalid}
-    isInline={isInline}
-    isLabelHidden={isLabelHidden}
-    htmlFor={selectId}
-    maxWidth={maxWidth}
-  >
-    <Select
-      {...selectAttrs}
-      id={selectId}
-      isInvalid={isInvalid}
-      defaultValue={defaultValue}
-      onChange={onChange}
-      placeholder={placeholder}
-      value={value}
-      variant={variant}
-    >
-      {children}
-    </Select>
-  </Field>
-);
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Component is internationalized, and must be wrapped in terra-base');
+    }
+  },
+};
+
+class SelectField extends React.Component {
+  constructor() {
+    super();
+
+    this.state = { help: undefined };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(value) {
+    const { maxSelectionCount, variant } = this.props;
+
+    if (maxSelectionCount !== undefined && MenuUtil.allowsMultipleSelections(variant) && Array.isArray(value) && value.length >= maxSelectionCount) {
+      let helpWithMaxSelectionCount = this.context.intl.formatMessage({ id: 'Terra.form.select.maxSelectionHelp' }, { text: maxSelectionCount });
```

Comment:
pretty sure we want the `helpWithMaxSelectionCount` to be injected always, not conditional if maxSelectionCount is reached.

Reply:
Ok. I'm updating it to always show the "Limit x items" in the help text.

Reply:
Updated: https://github.com/cerner/terra-core/pull/2244/commits/3c8375b9ae7e72f9eb55a40a15eab66819461ff7